docs/docs/workflows/events/concurrency-and-throttling/README.md

```diff
@@ -0,0 +1,67 @@
+
+**Concurrency**
+
+Without restricting concurrency, events can be processed in parallel and there is no guarantee that they will execute in the order in which they were received. This can cause race conditions. 
+
+For example, if two workflow events try to add data to Google Sheets simultaneously, they may both attempt to write data to the same row. As a result, one event can overwrite data from another event. The diagram below illustrates this example -- both `Event 1` and `Event 2` attempt to write data to Google Sheets concurrently -- as a result, they will both write to the same row and the data for one event will be overwritten and lost (and no error will be thrown). We observed this scenario resulted in data loss approximately 20% of the time with high volume workflows.
+
+![image-20201027132901691](images/image-20201027132901691.png)
+
+You can avoid race conditions like this by limiting workflow concurrency to a single "worker". What this means is that only one event will be processed at a time, and the next event will not start processing until the first is complete (unprocesssed events will maintained in a queue and processed by the workflow in order). The following diagram illustrates how the events in the last diagram would executed if concurrency was limited to a single worker. 
+
+![image-20201027133308888](images/image-20201027133308888.png)
+
+While the first example resulted in only two rows of data in Google Sheets, this time data for all three events are recorded to three separate rows.
+
+**Throttling**
+
+If your workflow integrates with any APIs, then you may need to limit the rate at which your workflow executes to avoid hitting rate limits from your API provider. Since event-driven workflows are stateless, you can't manage the rate of execution from within your workflow code. Pipedream's execution controls solve this problem by allowing you to control the maximum number of times a workflow may be invoked over a specific period of time (e.g., up to 1 event every second).
+
+## How It Works
+
+Events emitted from a source to a workflow are placed in a queue, and Pipedream triggers your workflow with events from the queue based on your concurrency and rate limit settings. These settings may be customized per workflow (so the same events may be processed at different rates by different workflows).
+
+![image-20201027145847752](images/image-20201027145847752.png)
+
+The maximum number of events Pipedream will queue per workflow depends on your account type.
+
+- Up to 100 events will be queued per workflow for free and pro accounts
+- Team/Enterprise accounts may have custom limits. If you need a larger queue size, please contact Pipedream
+
+**IMPORTANT:** If the number of events emitted to a workflow exceeds the queue size, events will be lost. If that happens, an error message will be displayed in the event list of your workflow and your global error workflow will be triggered.
```

Comment:
it's a Queue Size Exceeded limit right? We should probably specify that and add that error to https://docs.pipedream.com/errors/#pipedream-internal-errors